test/common/http/http1/codec_impl_test.cc

```diff
@@ -349,10 +371,19 @@ TEST_F(Http1ServerConnectionImplTest, IdentityAndChunkedBody) {
   MockRequestDecoder decoder;
   EXPECT_CALL(callbacks_, newStream(_, _)).WillOnce(ReturnRef(decoder));
 
+  TestHeaderMapImpl expected_headers{
+      {":path", "/"},
+      {":method", "POST"},
+      {"transfer-encoding", "identity, chunked"},
+  };
+  EXPECT_CALL(decoder, decodeHeaders_(HeaderMapEqual(&expected_headers), false)).Times(1);
+  Buffer::OwnedImpl expected_data("Hello World");
+  EXPECT_CALL(decoder, decodeData(BufferEqual(&expected_data), false)).Times(1);
+  EXPECT_CALL(decoder, decodeData(_, true)).Times(1);
   Buffer::OwnedImpl buffer("POST / HTTP/1.1\r\ntransfer-encoding: "
-                           "identity,chunked\r\n\r\nb\r\nHello World\r\n0\r\n\r\n");
-  EXPECT_THROW_WITH_MESSAGE(codec_->dispatch(buffer), CodecProtocolException,
-                            "http/1.1 protocol error: unsupported transfer encoding");
+                           "identity, chunked\r\n\r\nb\r\nHello World\r\n0\r\n\r\n");
+  codec_->dispatch(buffer);
+  EXPECT_EQ(0U, buffer.length());
```

Comment:
Should we continue disallowing identity,chunked

Reply:
I don't see what the harm is in allowing it since it doesn't change the handling of the request compared to just `chunked`. AFAIK the only reason it was disallowed earlier was because the parser didn't support multiple encodings, which it now does with this update. But if we want to continue disallowing it, that simplifies the `IsSupportedTransferEncoding` to just matching `chunked`.

Reply:
The argument I see for disallowing it is that it would protect servers behind Envoy that don't handle identity,chunked correctly.  We make the world slightly more secure by rejecting these requests.  I don't know if there is a significant rate of requests in the wild that we are incorrectly rejecting due to not supporting this transfer-encoding combination.

What's your preference?  I think we could go either way here.

Reply:
I don't have a strong preference either since I personally haven't seen much nor make use of `identity` or `identity, chunked`.

My initial reaction is to try to conform to RFC as much as possible as that results in the least surprises for people observing the behavior. Multiple encodings has been around since RFC 2616 (as well as `identity`), plus transfer-encoding is considered a "hop-by-hop" header, so the upstream request could theoretically just be `chunked` if needed (I doubt it'd be necessary, but is possible as a feature request for misbehaving servers?). So I suppose I'm leaning towards allowing `identity, chunked`. As I mentioned, I don't think disallowing `identity, chunked` had anything to do with `identity` or `chunked` specifically, but was just the global disallowing of `<encoding>, <encoding>` due to lacking library support which is resolved here.

Reply:
Sounds good, let's add support for "identity, chunked".  Do see Alyssa's request to separate pure import from behavior changes.